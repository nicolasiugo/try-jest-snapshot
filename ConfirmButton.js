import React, {Component, PropTypes} from "react";
import classNames from "classnames";

class ConfirmButton extends Component {

    constructor(props) {
        super(props);
        this.onAction = this.onAction.bind(this);
        this.onCancel = this.onCancel.bind(this);
        this.onAccept = this.onAccept.bind(this);
        this.state = {
            isConfirming: false
        }
    }

    onAction(e) {
        e.stopPropagation();
        e.cancelBubble = true; //IE
        this.setState({
            isConfirming: true
        });
    }

    onCancel(e) {
        e.stopPropagation();
        e.cancelBubble = true; //IE
        this.setState({
            isConfirming: false
        });
    }

    onAccept(e) {
        e.stopPropagation();
        e.cancelBubble = true; //IE
        this.setState({
            isConfirming: false
        });
        this.props.action();
    }


    render() {
        const {label, actionClassNames, confirmClassNames} = this.props;
        const actionClass = classNames({ 'hide': this.state.isConfirming }) + ' ' + actionClassNames;
        const confirmClass = classNames({ 'hide': !this.state.isConfirming }) + ' ' + confirmClassNames;
        return (
            <div className="confirm-button">
                <div
                    className={actionClass}
                    onClick={this.onAction}>
                  {label}
                </div>
                <div className={confirmClass}>
                    Are you sure? <a onClick={this.onAccept}>Yes</a> <a onClick={this.onCancel}>Cancel</a>
                </div>
            </div>
        );
    }
}

ConfirmButton.propTypes = {
    label: PropTypes.string.isRequired,
    action: PropTypes.func.isRequired,
    actionClassNames: PropTypes.string,
    confirmClassNames: PropTypes.string
};

export default ConfirmButton;
