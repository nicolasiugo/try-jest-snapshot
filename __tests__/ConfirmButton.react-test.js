'use strict';

import React from 'react';
import ConfirmButton from '../ConfirmButton';
import renderer from 'react/lib/ReactTestRenderer';

describe('ConfirmButton', () => {
  it('should render ConfirmButton component', () => {
    const actionSpy = jasmine.createSpy();
    const tree = renderer.create(
      <ConfirmButton
        action={actionSpy}
        confirmClassNames="confirm_class"
        actionClassNames="action_class"
        label="Important action"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('changes the class when clicked', () => {
    const actionSpy = jasmine.createSpy();
    const component = renderer.create(
      <ConfirmButton
        action={actionSpy}
        confirmClassNames="confirm_class"
        actionClassNames="action_class"
        label="Important action"
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    console.log(tree.props);
    tree.children[0].props.onClick({stopPropagation: jasmine.createSpy()});
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

  });

});