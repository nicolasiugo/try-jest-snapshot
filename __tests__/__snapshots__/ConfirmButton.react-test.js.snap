exports[`ConfirmButton changes the class when clicked 1`] = `
<div
  className="confirm-button">
  <div
    className=" action_class"
    onClick={[Function bound onAccept]}>
    Important action
  </div>
  <div
    className="hide confirm_class">
    Are you sure? 
    <a
      onClick={[Function bound onAccept]}>
      Yes
    </a>
     
    <a
      onClick={[Function bound onAccept]}>
      Cancel
    </a>
  </div>
</div>
`;

exports[`ConfirmButton changes the class when clicked 2`] = `
<div
  className="confirm-button">
  <div
    className="hide action_class"
    onClick={[Function bound onAccept]}>
    Important action
  </div>
  <div
    className=" confirm_class">
    Are you sure? 
    <a
      onClick={[Function bound onAccept]}>
      Yes
    </a>
     
    <a
      onClick={[Function bound onAccept]}>
      Cancel
    </a>
  </div>
</div>
`;

exports[`ConfirmButton should render ConfirmButton component 1`] = `
<div
  className="confirm-button">
  <div
    className=" action_class"
    onClick={[Function bound onAccept]}>
    Important action
  </div>
  <div
    className="hide confirm_class">
    Are you sure? 
    <a
      onClick={[Function bound onAccept]}>
      Yes
    </a>
     
    <a
      onClick={[Function bound onAccept]}>
      Cancel
    </a>
  </div>
</div>
`;
